### PR TITLE
CI: Automatically trigger framework build by tag

### PR DESCRIPTION
**Pull request by bisgardo**
To reduce the number of steps (and things to do wrong) in the release process, the workflow is updated to automatically trigger once a tag of form `build/*.*.*-*` is pushed. The workflow will then extract the version and run by itself.

A check that the version embedded in the tag actually matches the version of the crate was also included.

### CODE REVIEW

**Comment by bisgardo** [comment]
@soerenbf This is an attempt to make the release process a little easier. Do you think it's worth adding?

File: .github/workflows/publish-framework.yml [hunk 1]
```diff
@@ -16,12 +18,37 @@ jobs:
...
     - name: Setup Rust
       run: make setup
+    - name: Extract version of the crate
+      uses: bisgardo/github-action-echo@v1
+      id: crate-version
+      with:
+        version: |
+          $(cargo metadata --format-version=1 | jq -r '.packages[] | select(.name == "concordium-wallet-crypto-uniffi") | .version')
+    - name: Fail if version in tag doesn't match crate version
+      if: "steps.crate-version.version != steps.tag.outputs.version"
```
Comment (bisgardo) [inline .github/workflows/publish-framework.yml]:
```suggestion
      if: "steps.crate-version.outputs.version != steps.tag.outputs.version"
```

File: README.md [hunk 2]
```diff
@@ -125,21 +125,21 @@ Steps for building and releasing a new version `<version>` of the package:
    where `<build-version>` starts out at `0`.
    Tagging is necessary because the workflow to be run in the next step uploads a release,
    and GitHub requires releases to be tagged.
-4. Run the [workflow](./.github/workflows/publish-framework.yml) for publishing a new version of the binary framework.
-   Use the tag you just created as "branch" to run from and input `<version>` (i.e. without the build number).
-   If the build fails, commit a fix to the branch and return to step 3 with `<build-version>` incremented by 1.
-5. Run `make swift-bindings` locally to regenerate the Swift bridge sources.
-6. Update `Package.swift` with the updated `url` and `checksum` of the binary framework.
+   The pushing of this tag will automatically trigger a [workflow](./.github/workflows/publish-framework.yml)
```
Comment (soerenbf) [inline README.md]:
```suggestion
   Pushing this tag will automatically trigger a [workflow](./.github/workflows/publish-framework.yml)
```

**Review by soerenbf — APPROVED** [review]
Looks good 😄 . I think (as a comment to the description) that if we have build tags specifically for the purpose of doing this thing, we might as well automate. I still think there are a lot of quirky steps to follow in order to do this but if there is no way around it, I guess it is what it is 🤷

**Comment by bisgardo** [comment]
> Looks good 😄 . I think (as a comment to the description) that if we have build tags specifically for the purpose of doing this thing, we might as well automate. I still think there are a lot of quirky steps to follow in order to do this but if there is no way around it, I guess it is what it is 🤷

It could probably be wrapped into fewer, more customized actions. But it's probably quite a lot of work to do in a good way.

Also the "tag override" think isn't really necessary, so I can simplify it a little by removing that.

**Pull request merged**